docs/criteria/connectors/rc.md

```diff
@@ -0,0 +1,143 @@
+# Spice.ai OSS Data Connectors - RC Release Criteria
+
+This document defines the set of criteria that is required before a data connector is considered to be of [RC](../definitions.md) quality.
+
+All criteria must be met for the connector to be considered [RC](../definitions.md), with exceptions only permitted in some circumstances (e.g. it would be technically infeasible to add a feature/fix a bug for a particular connector).
+
+## RC Quality Connectors
+
+| Connector | RC Quality | DRI Sign-off |
+| - | - | - |
+| Clickhouse    | ❌ |  |
+| Databricks    | ❌ |  |
+| Delta Lake    | ❌ |  |
+| Dremio        | ❌ |  |
+| File          | ❌ |  |
+| FTP/SFTP      | ❌ |  |
+| GraphQL       | ❌ |  |
+| GitHub        | ❌ |  |
+| HTTP/HTTPS    | ❌ |  |
+| MS SQL        | ❌ |  |
+| MySQL         | ❌ |  |
+| ODBC          | ❌ |  |
+| PostgreSQL    | ❌ |  |
+| Sharepoint    | ❌ |  |
+| Snowflake     | ❌ |  |
+| Spice.AI      | ❌ |  |
+| S3            | ❌ |  |
+| Azure BlobFS  | ❌ |  |
+| Spark         | ❌ |  |
+
+## RC Release Criteria
+
+Some connectors impose different testing requirements, as by their nature they support different feature requirements.
+
+Some connectors are unable to support TPC derived test packages due to their nature (e.g. GitHub Issues).
+These connectors are exempt from running the TPC derived test packages, and rely instead on their general integration test.
+
+This table defines the required features and/or tests for each connector:
+
+| Connector     | [TPC-H Derived Tests (Scale Factor)](#tpc-h) | [TPC-DS Derived Tests (Scale Factor)](#tpc-ds) | [Federation](#federation) | [Data Correctness](#data-correctness) | [Streaming](#streaming) | [Native Schema Inference](#schema-inference) |
+| ------------- | - | - | - | - | - | - |
+| Clickhouse    | ✅ (100) | ✅ (100) | ✅ | ✅ | ✅ | ✅ |
+| Databricks    | ✅ (100) | ✅ (100) | ✅ | ✅ | ✅ | ✅ |
+| Delta Lake    | ✅ (100) | ✅ (100) | ✅ | ✅ | ✅ | ✅ |
+| Dremio        | ✅ (100) | ✅ (100) | ✅ | ✅ | ✅ | ✅ |
+| File          | ✅ (1) | ✅ (0.5) | ❌ | ❌ | ❌ | ❌ |
+| FTP/SFTP      | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
```

Comment:
I think some of these support streaming, in the definition of ~ not all rows of data needs to be retrieved before being provided to a downstream component of the system reading the data (still separate from whether the connector supports `LIMIT` pushdown). In this case, I think the following support streaming:
 - File/S3/Azure BlobFS: For document table formats, each file is streamed individually, and the _"list metadata"_ is [streamed](https://github.com/spiceai/spiceai/blob/6534930313a0fe6cba1e4fa96bdbfcdc4259b529/crates/data_components/src/object/text.rs#L265). For tabular data (e.g), it is determined per file format, but generally i believe they also stream the rows within a file, e.g. [csv](https://github.com/apache/datafusion/blob/1b10c9f89eac127507fe7a137ff7c40534f7ca9a/datafusion/core/src/datasource/file_format/csv.rs#L145-L150).